---
title: "PRS BrainAge"
author: "Jorge Garcia Condado"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_float: true
knit: (
  function(inputFile, encoding) { 
    rmarkdown::render('PRSBrainAge.Rmd',
      output_file = paste('~/Documents/Projects/PRSBrainAgePlasma/PRSBrainAge', '.html', sep=''))
      })
---

```{r, include = FALSE, warning = FALSE, message = FALSE}
library(ggplot2); library(ggpubr); library(dplyr);library(sjPlot);library(gtsummary); library(flextable);library(ppcor);library(gridExtra);library(effsize)


knitr::opts_chunk$set(echo = FALSE, message = FALSE)
theme_set(theme_bw())
```

# Polygenetic Risk Score of BrainAge

To examine the association between genetic predisposition to accelerated brain aging—measured by polygenic risk scores (PRS) derived from BrainAge models—and plasma biomarkers of Alzheimer’s disease (AD), with attention to age and sex-specific effects.

```{r, include = FALSE}

# Load PRS
threshold = 0.5
prs_gm <- read.table(sprintf('~/Documents/Projects/PRSBrainAgePlasma/data/prs_scores/gm/prs2.pT%s.sscore', threshold), 
                     header = FALSE, 
                     row.names = 1,
                     sep = '\t') %>%
          dplyr::select(V2, V5) %>%
          rename(BID = V2,GM_SCORE= V5)
prs_wm <- read.table(sprintf('~/Documents/Projects/PRSBrainAgePlasma/data/prs_scores/wm/prs2.pT%s.sscore', threshold), 
                     header = FALSE, 
                     row.names = 1,
                     sep = '\t') %>%
          dplyr::select(V2, V5) %>%
          rename(BID = V2, WM_SCORE= V5)
prs_fc <- read.table(sprintf('~/Documents/Projects/PRSBrainAgePlasma/data/prs_scores/fc/prs2.pT%s.sscore', threshold), 
                     header = FALSE, 
                     row.names = 1,
                     sep = '\t') %>%
          dplyr::select(V2, V5) %>%
          rename(BID = V2, FC_SCORE= V5)

# Merge PRS
prs <- merge(prs_gm, prs_wm, by = 'BID')
prs <- merge(prs, prs_fc, by = 'BID')

# Z-score PRS
prs$GM_ZSCORE <- (prs$GM_SCORE - mean(prs$GM_SCORE, na.rm = TRUE)) / sd(prs$GM_SCORE, na.rm = TRUE)
prs$WM_ZSCORE <- (prs$WM_SCORE - mean(prs$WM_SCORE, na.rm = TRUE)) / sd(prs$WM_SCORE, na.rm = TRUE)
prs$FC_ZSCORE <- (prs$FC_SCORE - mean(prs$FC_SCORE, na.rm = TRUE)) / sd(prs$FC_SCORE, na.rm = TRUE)

# Load data
data <- read.csv('~/Documents/Projects/PRSBrainAgePlasma/data/a4_baseline_plasma/a4_plasma_05012025_full.csv', 
                 row.names=1)  %>%
        dplyr::select(BID, Sex, Baseline_Age, Education, apoe_genotype, Amyloid_Status, PTAU217, NF.L., GFAP.) %>%
        rename(age = Baseline_Age, YearsEd = Education, Amyloid = Amyloid_Status, pTau217 = PTAU217, NF.L = NF.L., GFAP = GFAP.) %>%
        mutate(Sex = recode(Sex, `1` = "Female", `0` = "Male"),
               Sex = as.factor(Sex),
               Amyloid = as.factor(Amyloid))

# Filter value reads of PTau217
data$pTau217[data$pTau217 == "<LLOQ" | data$pTau217 == ">ULOQ"] <- NA
data$pTau217 <- as.numeric(data$pTau217)
data$NF.L <- as.numeric(data$NF.L)

# For numbering puprose
data$ptau_n <- ifelse(!is.na(data$pTau217), 1, 0)
data$NF.L_n <- ifelse(!is.na(data$NF.L), 1, 0)
data$GFAP_n <- ifelse(!is.na(data$GFAP), 1, 0)

# Remove those that have NF.L 1 but GFAP 0 or GFAP 1 but NF.L 0 (they all come from the same platform)
data <- data[!(data$NF.L_n == 1 & data$GFAP_n == 0), ]
data <- data[!(data$GFAP_n == 1 & data$NF.L_n == 0), ]

# Remove those that have 0 for pTau217, GFAP, NFL
data <- data[!(data$pTau217 == 0 & data$GFAP == 0 & data$NF.L == 0), ]

# APOE e4 status
data$e4 <- ifelse(grepl("e4", data$apoe_genotype), "e4+", "e4-")
data$e4 <- as.factor(data$e4)

# Merge PRS
data <- merge(data, prs, by = 'BID')
```

## Cohort Demographics

Data form the A4/LEARN study was used. Plasma samples were processed by two different companies: Eli Lilly and Company (Lilly) to develop a pTau217 assay and Roche Diagnostic for testing of the Elecsys Robust Prototype Immunoassays.

```{r}
# Cross tabulate those with NfL and ptau217
t_ptau_gfap <- table(data$ptau_n, data$GFAP_n)
# Add titles to the rows and columns
dimnames(t_ptau_gfap) <- list(
  "pTau217" = c("NA", "Measurment"),
  "GFAP" = c("NA", "Measurment")
)
t_ptau_gfap

```

```{r}
table <- tbl_summary(data,
            include = c(age, Sex, YearsEd, e4, Amyloid, ptau_n, pTau217, GFAP_n, NF.L, GFAP),
            missing = "no",
            statistic = list(all_continuous()  ~ c("{mean} ({sd})")),
            label = list(age ~ "Age", YearsEd ~ "Education (Years)", NF.L ~ "NfL (pg/mL)", GFAP ~ "GFAP (ng/mL)", pTau217 ~ "pTau217 (U/mL)", ptau_n ~ "Lilly Platform (N)", GFAP_n ~ "Roche Platform (N)"),
            digits = list(all_continuous() ~ 2, all_categorical() ~ c(0,2)),
            by = Amyloid) %>%
  modify_header(stat_1 = "**AB-**, N = {n}", stat_2 = "**AB+**, N = {n}") %>%
  add_p() %>%
  add_overall() %>%
  bold_labels() %>%
  modify_table_styling(
          columns = label,
          rows = label == "pTau217 (U/mL)",
          footnote = "Lilly Clinical Diagnostics Laboratory conducted pTau217 testing using an automated electrochemiluminescent immunoassay (Tecan Fluent workstation for preparation, MSD Sector S Imager 600MM for detection)",
        ) %>%
  modify_table_styling(
          columns = label,
          rows = label == "GFAP (ng/mL)",
          footnote = "Roche Diagnostics conducted testing of their Elecsys Robust Prototype Immunoassays.",
        ) %>%
  modify_table_styling(
          columns = label,
          rows = label == "NfL (pg/mL)",
          footnote = "Roche Diagnostics conducted testing of their Elecsys Robust Prototype Immunoassays.",
        )

table %>%
  as_flex_table() %>% 
  fontsize(size = 12, part = "all") %>%
  line_spacing(space = 1.5, part = "all") %>%
  autofit() %>%
  save_as_docx(path = "~/Documents/Projects/PRSBrainAgePlasma/figures/Cohort_Demographics.docx")

table

```

## Distribution of PRS Z-scores

PRS were computed for three BrainAge models—Grey Matter (GM), White Matter (WM), and Functional Connectivity (FC)—using GWAS summary statistics from [Wen et al., 2024](https://www.nature.com/articles/s41467-024-46796-6). 

```{r}
# Plot three histograms of PRS in a grid
h_gm <- ggplot(data, aes(x = GM_ZSCORE)) +
  geom_histogram(binwidth = 0.5, fill = "blue", color = "black", alpha = 0.7) +
  labs(title = "GM PRS", x = "GM PRS Z-score", y = "Count") +
  theme_minimal()
h_wm <- ggplot(data, aes(x = WM_ZSCORE)) +
  geom_histogram(binwidth = 0.5, fill = "red", color = "black", alpha = 0.7) +
  labs(title = "WM PRS", x = "WM PRS Z-score", y = "Count") +
  theme_minimal()
h_fc <- ggplot(data, aes(x = FC_ZSCORE)) +
  geom_histogram(binwidth = 0.5, fill = "green", color = "black", alpha = 0.7) +
  labs(title = "FC PRS", x = "FC PRS Zscore", y = "Count") +
  theme_minimal()
grid.arrange(h_gm, h_wm, h_fc, ncol = 3)
```

# Group analysis {.tabset}

Difference in PRS scores for different covariates.

## Sex {.tabset}

```{r}
# Create a bloxpot groupbing by SEX
sex_gm <- ggplot(subset(data, !is.na(Sex)), aes(x = Sex, y = GM_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'WM PRS',
       x = 'SEX',
       y = 'PRS Zscore')

# Create a bloxpot groupbing by SEX
sex_wm <- ggplot(subset(data, !is.na(Sex)), aes(x = Sex, y = WM_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'GM PRS',
       x = 'SEX',
       y = 'PRS Zscore')

# Create a bloxpot groupbing by SEX
sex_fc <- ggplot(subset(data, !is.na(Sex)), aes(x = Sex, y = FC_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'FC PRS',
       x = 'SEX',
       y = 'PRS Zscore')

# Arrange the plots in a grid
grid.arrange(sex_gm, sex_wm, sex_fc, ncol = 3)
```

## E4 Status

```{r}
# Create a boxplot grouping by E4 status
e4_gm <- ggplot(subset(data, !is.na(e4)), aes(x = e4, y = GM_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'GM PRS',
       x = 'E4 status',
       y = 'PRS Zscore')

# Create a boxplot grouping by E4 status
e4_wm <- ggplot(subset(data, !is.na(e4)), aes(x = e4, y = WM_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'WM PRS',
       x = 'E4 status',
       y = 'PRS Zscore')

# Create a boxplot grouping by E4 status
e4_fc <- ggplot(subset(data, !is.na(e4)), aes(x = e4, y = FC_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'FC PRS',
       x = 'E4 status',
       y = 'PRS Zscore')

# Arrange the plots in a grid
grid.arrange(e4_gm, e4_wm, e4_fc, ncol = 3)
```

## Amyloid Status

```{r}
# Create a boxplot grouping by Amyloid status

ab_gm <- ggplot(subset(data, !is.na(Amyloid)), aes(x = Amyloid, y = GM_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'GM PRS',
       x = 'Amyloid status',
       y = 'PRS Zscore')

# Create a boxplot grouping by Amyloid status
ab_wm <- ggplot(subset(data, !is.na(Amyloid)), aes(x = Amyloid, y = WM_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'WM PRS',
       x = 'Amyloid status',
       y = 'PRS Zscore')

# Create a boxplot grouping by Amyloid status
ab_fc <- ggplot(subset(data, !is.na(Amyloid)), aes(x = Amyloid, y = FC_ZSCORE)) +
  geom_boxplot() +
  stat_compare_means(method = 't.test') +
  labs(title = 'FC PRS',
       x = 'Amyloid status',
       y = 'PRS Zscore')

# Arrange the plots in a grid
grid.arrange(ab_gm, ab_wm, ab_fc, ncol = 3)
```

## Age

``` {r}
# Create a scatter plot of PRS Zscore by Age
age_gm <- ggplot(subset(data, !is.na(age)), aes(x = age, y = GM_ZSCORE)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'GM PRS',
       x = 'Age (years)',
       y = 'PRS Zscore')

# Create a scatter plot of PRS Zscore by Age
age_wm <- ggplot(subset(data, !is.na(age)), aes(x = age, y = WM_ZSCORE)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'WM PRS',
       x = 'Age (years)',
       y = 'PRS Zscore')

# Create a scatter plot of PRS Zscore by Age
age_fc <- ggplot(subset(data, !is.na(age)), aes(x = age, y = FC_ZSCORE)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'PRS Zscore by Age',
       x = 'Age (years)',
       y = 'PRS Zscore')

# Arrange the plots in a grid
grid.arrange(age_gm, age_wm, age_fc, ncol = 3)
```

## Education

``` {r}
# Create a scatter plot of PRS Zscore by Age
edu_gm <- ggplot(subset(data, !is.na(YearsEd)), aes(x = YearsEd, y = GM_ZSCORE)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'GM PRS',
       x = 'Education (years)',
       y = 'PRS Zscore')

# Create a scatter plot of PRS Zscore by Age
edu_wm <- ggplot(subset(data, !is.na(YearsEd)), aes(x = YearsEd, y = WM_ZSCORE)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'WM PRS',
       x = 'Education (years)',
       y = 'PRS Zscore')

# Create a scatter plot of PRS Zscore by Years Education
edu_fc <- ggplot(subset(data, !is.na(YearsEd)), aes(x = YearsEd, y = FC_ZSCORE)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'FC PRS',
       x = 'Education (years)',
       y = 'PRS Zscore')

# Arrange the plots in a grid
grid.arrange(edu_gm, edu_wm, edu_fc, ncol = 3)
```

# General Linear Models for plasma biomaker {.tabset}

We build a general linear model that predicts plasma biomarkers using PRS, Age and PRS\*Age interaction

## pTau217

```{r}
# GM
ptau_gm <- glm(pTau217 ~ GM_ZSCORE*age, data = data)
ptau_gm_standardized <- effectsize::standardize(ptau_gm, standardize = "all")
ptau_gm_results <- summary(ptau_gm_standardized)$coefficients
cat("pTau217 GM PRS \n")
cat("Standardized Effect GM PRS * Age: ", ptau_gm_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", ptau_gm_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", ptau_gm_results["GM_ZSCORE", "Estimate"], 
    "p-val", ptau_gm_results["GM_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", ptau_gm_results["age", "Estimate"], 
    "p-val", ptau_gm_results["age", "Pr(>|t|)"], "\n")

# WM
ptau_wm <- glm(pTau217 ~ WM_ZSCORE*age, data = data)
ptau_wm_standardized <- effectsize::standardize(ptau_wm, standardize = "all")
ptau_wm_results <- summary(ptau_wm_standardized)$coefficients
cat("pTau217 WM PRS \n")
cat("Standardized Effect WM PRS * Age: ", ptau_wm_results["WM_ZSCORE:age", "Estimate"], 
    "p-val", ptau_wm_results["WM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect WM PRS: ", ptau_wm_results["WM_ZSCORE", "Estimate"], 
    "p-val", ptau_wm_results["WM_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", ptau_wm_results["age", "Estimate"], 
    "p-val", ptau_wm_results["age", "Pr(>|t|)"], "\n")

# FC
ptau_fc <- glm(pTau217 ~ FC_ZSCORE*age, data = data)
ptau_fc_standardized <- effectsize::standardize(ptau_fc, standardize = "all")
ptau_fc_results <- summary(ptau_fc_standardized)$coefficients
cat("pTau217 FC PRS \n")
cat("Standardized Effect FC PRS * Age: ", ptau_fc_results["FC_ZSCORE:age", "Estimate"], 
    "p-val", ptau_fc_results["FC_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect FC PRS: ", ptau_fc_results["FC_ZSCORE", "Estimate"], 
    "p-val", ptau_fc_results["FC_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", ptau_fc_results["age", "Estimate"], 
    "p-val", ptau_fc_results["age", "Pr(>|t|)"], "\n")
```

## GFAP

```{r}
# GM
gfap_gm <- glm(GFAP ~ GM_ZSCORE*age, data = data)
gfap_gm_standardized <- effectsize::standardize(gfap_gm, standardize = "all")
gfap_gm_results <- summary(gfap_gm_standardized)$coefficients
cat("GFAP GM PRS \n")
cat("Standardized Effect GM PRS * Age: ", gfap_gm_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", gfap_gm_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", gfap_gm_results["GM_ZSCORE", "Estimate"], 
    "p-val", gfap_gm_results["GM_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", gfap_gm_results["age", "Estimate"], 
    "p-val", gfap_gm_results["age", "Pr(>|t|)"], "\n")

# WM
gfap_wm <- glm(GFAP ~ WM_ZSCORE*age, data = data)
gfap_wm_standardized <- effectsize::standardize(gfap_wm, standardize = "all")
gfap_wm_results <- summary(gfap_wm_standardized)$coefficients
cat("GFAP WM PRS \n")
cat("Standardized Effect WM PRS * Age: ", gfap_wm_results["WM_ZSCORE:age", "Estimate"], 
    "p-val", gfap_wm_results["WM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect WM PRS: ", gfap_wm_results["WM_ZSCORE", "Estimate"], 
    "p-val", gfap_wm_results["WM_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", gfap_wm_results["age", "Estimate"], 
    "p-val", gfap_wm_results["age", "Pr(>|t|)"], "\n")

# FC
gfap_fc <- glm(GFAP ~ FC_ZSCORE*age, data = data)
gfap_fc_standardized <- effectsize::standardize(gfap_fc, standardize = "all")
gfap_fc_results <- summary(gfap_fc_standardized)$coefficients
cat("GFAP FC PRS \n")
cat("Standardized Effect FC PRS * Age: ", gfap_fc_results["FC_ZSCORE:age", "Estimate"], 
    "p-val", gfap_fc_results["FC_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect FC PRS: ", gfap_fc_results["FC_ZSCORE", "Estimate"], 
    "p-val", gfap_fc_results["FC_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", gfap_fc_results["age", "Estimate"], 
    "p-val", gfap_fc_results["age", "Pr(>|t|)"], "\n")
```

## NFL

```{r}
# GM
nfl_gm <- glm(NF.L ~ GM_ZSCORE*age, data = data)
nfl_gm_standardized <- effectsize::standardize(nfl_gm, standardize = "all")
nfl_gm_results <- summary(nfl_gm_standardized)$coefficients
cat("NF.L GM PRS \n")
cat("Standardized Effect GM PRS * Age: ", nfl_gm_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", nfl_gm_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", nfl_gm_results["GM_ZSCORE", "Estimate"], 
    "p-val", nfl_gm_results["GM_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", nfl_gm_results["age", "Estimate"], 
    "p-val", nfl_gm_results["age", "Pr(>|t|)"], "\n")

# WM
nfl_wm <- glm(NF.L ~ WM_ZSCORE*age, data = data)
nfl_wm_standardized <- effectsize::standardize(nfl_wm, standardize = "all")
nfl_wm_results <- summary(nfl_wm_standardized)$coefficients
cat("NF.L WM PRS \n")
cat("Standardized Effect WM PRS * Age: ", nfl_wm_results["WM_ZSCORE:age", "Estimate"], 
    "p-val", nfl_wm_results["WM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect WM PRS: ", nfl_wm_results["WM_ZSCORE", "Estimate"], 
    "p-val", nfl_wm_results["WM_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", nfl_wm_results["age", "Estimate"], 
    "p-val", nfl_wm_results["age", "Pr(>|t|)"], "\n")

# FC
nfl_fc <- glm(NF.L ~ FC_ZSCORE*age, data = data)
nfl_fc_standardized <- effectsize::standardize(nfl_fc, standardize = "all")
nfl_fc_results <- summary(nfl_fc_standardized)$coefficients
cat("NF.L FC PRS \n")
cat("Standardized Effect FC PRS * Age: ", nfl_fc_results["FC_ZSCORE:age", "Estimate"], 
    "p-val", nfl_fc_results["FC_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect FC PRS: ", nfl_fc_results["FC_ZSCORE", "Estimate"], 
    "p-val", nfl_fc_results["FC_ZSCORE", "Pr(>|t|)"], "\n")
cat("Standardized Effect Age: ", nfl_fc_results["age", "Estimate"], 
    "p-val", nfl_fc_results["age", "Pr(>|t|)"], "\n")
```

# pTau217 and GM BrainAge PRS

We only find significant results when looking at pTau217 and the effect of GM BrainAge PRS.

```{r}
ggplot(subset(data, !is.na(pTau217)), aes(x = GM_ZSCORE, y = pTau217)) +
  geom_point() +
  geom_smooth(method = 'lm', formula = y ~ x)+
  stat_cor() +
  labs(title = 'Comparison of PRS to pTau217 at Baseline',
       x = 'PRS Zscore',
       y = 'pTau217')
```

```{r}
# Plot OLS model
plot_model(ptau_gm, type = c('pred'), terms = c('age', 'GM_ZSCORE [-1, 0, 1]'), show.values = TRUE)
```


# Sensitivity Analysis 

We investigate how adding the other PRS, covariates and other plasma biomarkers affects the results.

```{r}
# Sensitivity analysis adding other PRS
ptau217_allprs <- glm(pTau217 ~ GM_ZSCORE*age + WM_ZSCORE*age + FC_ZSCORE*age, data = data)
ptau217_allprs_standardized <- effectsize::standardize(ptau217_allprs, standardize = "all")
ptau217_allprs_results <- summary(ptau217_allprs_standardized)$coefficients
cat("Adding PRS \n")
cat("Standardized Effect GM PRS * Age: ", ptau217_allprs_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", ptau217_allprs_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", ptau217_allprs_results["GM_ZSCORE", "Estimate"],
    "p-val", ptau217_allprs_results["GM_ZSCORE", "Pr(>|t|)"], "\n")

# Sensitivity analysis adding other PRS and covariates
ptau217_allprs_covars <- glm(pTau217 ~ GM_ZSCORE*age + WM_ZSCORE*age + FC_ZSCORE*age + Sex + YearsEd + e4 + Amyloid, data = data)
ptau217_allprs_covars_standardized <- effectsize::standardize(ptau217_allprs_covars, standardize = "all")
ptau217_allprs_covars_results <- summary(ptau217_allprs_covars_standardized)$coefficients
cat("Adding PRS and covariates \n")
cat("Standardized Effect GM PRS * Age: ", ptau217_allprs_covars_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", ptau217_allprs_covars_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", ptau217_allprs_covars_results["GM_ZSCORE", "Estimate"],
    "p-val", ptau217_allprs_covars_results["GM_ZSCORE", "Pr(>|t|)"], "\n")

# Sensitivity analysis adding other PRS, covariates and other plasma biomarkers
pTau217_biomakers <- glm(pTau217 ~ GM_ZSCORE*age + WM_ZSCORE*age + FC_ZSCORE*age + Sex + YearsEd + e4 + Amyloid + GFAP + NF.L, data = data)
pTau217_biomakers_standardized <- effectsize::standardize(pTau217_biomakers, standardize = "all")
pTau217_biomakers_results <- summary(pTau217_biomakers_standardized)$coefficients
cat("Adding PRS, covariates and other blood biomarkers \n")
cat("Standardized Effect GM PRS * Age: ", pTau217_biomakers_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", pTau217_biomakers_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", pTau217_biomakers_results["GM_ZSCORE", "Estimate"],
    "p-val", pTau217_biomakers_results["GM_ZSCORE", "Pr(>|t|)"], "\n")
```

# Sex differences

Add sex interaction

```{r}
ptau_sex <- lm(pTau217 ~ GM_ZSCORE*age*Sex, data = data)
ptau_sex_standardized <- effectsize::standardize(ptau_sex, standarize="all")
ptau_sex_results <- summary(ptau_sex_standardized)$coefficients
cat("Adding GM PRS x Age X Sex interaction \n")
cat("Standardized Effect GM PRS * Age * Sex: ", ptau_sex_results["GM_ZSCORE:age:SexMale", "Estimate"], 
    "p-val", ptau_sex_results["GM_ZSCORE:age:SexMale", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS * Age: ", ptau_sex_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", ptau_sex_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
plot_model(ptau_sex, type = c('pred'), terms = c('age', 'GM_ZSCORE [-1, 0, 1]', 'Sex'), show.values = TRUE)
```

Stratified sex models

```{r}
ptau_male <- lm(pTau217 ~ GM_ZSCORE*age, data = subset(data, Sex == "Male"))
ptau_male_standardized <- effectsize::standardize(ptau_male, standardize = "all")
ptau_male_results <- summary(ptau_male_standardized)$coefficients
cat("Male only model\n")
cat("Standardized Effect GM PRS * Age: ", ptau_male_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", ptau_male_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", ptau_male_results["GM_ZSCORE", "Estimate"],
    "p-val", ptau_male_results["GM_ZSCORE", "Pr(>|t|)"], "\n")
plot_model(ptau_male, type = c('pred'), terms = c('age', 'GM_ZSCORE [-1, 0, 1]'), show.values = TRUE)
```

```{r}
ptau_female <- lm(pTau217 ~ GM_ZSCORE*age, data = subset(data, Sex == "Female"))
ptau_female_standardized <- effectsize::standardize(ptau_female, standardize = "all")
ptau_female_results <- summary(ptau_female_standardized)$coefficients
cat("Female only model\n")
cat("Standardized Effect GM PRS * Age: ", ptau_female_results["GM_ZSCORE:age", "Estimate"], 
    "p-val", ptau_female_results["GM_ZSCORE:age", "Pr(>|t|)"], "\n")
cat("Standardized Effect GM PRS: ", ptau_female_results["GM_ZSCORE", "Estimate"],
    "p-val", ptau_female_results["GM_ZSCORE", "Pr(>|t|)"], "\n")
plot_model(ptau_female, type = c('pred'), terms = c('age', 'GM_ZSCORE [-1, 0, 1]'), show.values = TRUE)
```
